src/lib/autoinstall/clients/inst_autoinit.rb

```diff
@@ -0,0 +1,368 @@
+
+      def self.run
+        new.run
+      end
+
+      def initialize
+        textdomain "autoinst"
+
+        Yast.include self, "autoinstall/autoinst_dialogs.rb"
+      end
+
+      def run
+        Yast::Console.Init
+
+        @help_text = _(
+          "<p>\nPlease wait while the system is prepared for autoinstallation.</p>\n"
+        )
+        @progress_stages = [
+          _("Probe hardware"),
+          _("Retrieve & Read Control File"),
+          _("Parse control file"),
+          _("Initial Configuration")
+        ]
+        @profileFetched = false
+
+        Yast::Progress.New(
+          _("Preparing System for Automatic Installation"),
+          "", # progress_title
+          6, # progress bar length
+          @progress_stages,
+          [],
+          @help_text
+        )
+        Yast::Progress.NextStage
+        Yast::Progress.Title(_("Preprobing stage"))
+        Yast::Builtins.y2milestone("pre probing")
+
+        @tmp = Yast::Convert.to_string(
+          Yast::SCR.Read(Yast::Path.new(".target.string"), "/etc/install.inf")
+        )
+        if !@tmp.nil? && Yast::Builtins.issubstring(Yast::Builtins.tolower(@tmp), "iscsi: 1")
+          Yast::WFM.CallFunction("inst_iscsi-client", [])
+        end
+
+        Yast::Progress.NextStep
+        Yast::Progress.Title(_("Probing hardware..."))
+        Yast::Builtins.y2milestone("Probing hardware...")
+
+        if !@profileFetched
+          # if profile is defined, first read it, then probe hardware
+          @autoinstall = Yast::SCR.Read(Yast::Path.new(".etc.install_inf.AutoYaST"))
+          if Yast::Mode.autoupgrade &&
+              !(!@autoinstall.nil? && Ops.is_string?(@autoinstall) &&
+                Yast::Convert.to_string(@autoinstall) != "")
+            Yast::AutoinstConfig.ParseCmdLine("file:///mnt/root/autoupg.xml")
+            Yast::AutoinstConfig.ProfileInRootPart = true
+          end
+
+          @ret = processProfile
+          return @ret if @ret != :ok
+        end
+
+        Yast::Progress.Finish
+
+        # when installing from the online installation medium we need to
+        # register the system earlier because the medium does not contain any
+        # repositories, we need the repositories from the registration server
+        if Y2Packager::MediumType.online? && !Yast::Mode.autoupgrade
+          # check that the registration section is defined and registration is enabled
+          reg_section = Yast::Profile.current.fetch(REGISTER_SECTION, {})
+          reg_enabled = reg_section["do_registration"]
+
+          if !reg_enabled
+            msg = _("Registration is mandatory when using the online " \
+              "installation medium. Enable registration in " \
+              "the AutoYaST profile or use full installation medium.")
+            Yast::Popup.LongError(msg) # No timeout because we are stopping the installation/upgrade.
+
+            return :abort
+          end
+
+          suse_register if !Mode.autoupgrade
```

Comment:
At this point, this condition should be already met (it is the second part of the condition above).

Reply:
good point. Will fix it.